server/src/OutputParser.ts

```diff
@@ -59,20 +63,33 @@ export class OutputParser {
   }
 
   reportProblems (): void {
+    if (_connection === null) {
```

Comment:
what about we replace `console.warn` with `logger.warn`?

What about we replace all of this with a function to check whether the connection is there or not, to reduce the amount of repetition?
``` typescript
const checkConnectionIsSet (connection: Connection | null): boolean => {
  if (connection === null) {
    // eslint-disable-next-line no-console
    console.warn('The LSP Connection is not set. Dropping messages')
    return false
  }
  return true
}
```

Reply:
I replaced `console` to `logger`, however, I still need `_connection` here to let TS correctly infer the type.
We could have another object that calls the `connection` functions instead of calling it within this `OutputParser`.